src/handler.py

```diff
@@ -51,6 +50,41 @@ class BadRequestException(Exception):
     pass
 
 
+def _is_ignore_log_file(key=None, regex_pattern=None):
+    """
+    This functions checks whether this log file should be ignored based on regex pattern.
+    """
+    if not regex_pattern:
+        regex_pattern = os.getenv("S3_IGNORE_PATTERN", "$^")
+
+    return bool(re.search(regex_pattern, key))
+
+
+def _isCloudTrail(key=None, regex_pattern=None):
+    """
+    This functions checks whether this log file is a CloudTrail log based on regex pattern.
+    """
+    if not regex_pattern:
+        regex_pattern = os.getenv(
+            "S3_CLOUDTRAIL_LOG_PATTERN", ".*CloudTrail.*\.json.gz$")
+
+    return bool(re.search(regex_pattern, key))
+
+def _convert_float(s):
+    try:
+        f = float(s)
+    except ValueError:
+        f = 1.5
+    return f
+
+def _get_batch_size_factor(batch_size_factor=None):
```

Comment:
Why doesn't the `BATCH_SIZE_FACTOR` returned from this function fall back to the value we had before if it's not provided as an environment variable?